packages/core/src/application_init.ts

```
@@ -68,6 +68,11 @@ export class ApplicationInitStatus {
         const initResult = this.appInits[i]();
         if (isPromise(initResult)) {
           asyncInitPromises.push(initResult);
+        } else if (isObservable(initResult)) {
+          // Note: `.toPromise()` will be deprecated in RxJS v7 and removed completely in later
+          // versions. Once RxJS is updated in the repo to v7, this code should be refactored and
+          // the `lastValueFrom` function should be used instead.
```

I thought this would be the case, but `lastValueFrom` is fundamentally different to `toPromise` w.r.t. its behavior for empty observables: it throws an `EmptyError` instead of returning `undefined`. For this use-case I believe it's desirable to also accept observables that only complete without ever having emitted a value, so `lastValueFrom` would not be correct here.

---

thanks, I have removed this comment here.